tensorflow_datasets/image/deeplesion.py

```diff
@@ -0,0 +1,206 @@
+    'https://nihcc.box.com/shared/static/shr998yp51gf2y5jj7jqxz2ht8lcbril.zip',
+    'https://nihcc.box.com/shared/static/kqg4peb9j53ljhrxe3l3zrj4ac6xogif.zip'
+]
+
+_CSV_URL = "https://raw.githubusercontent.com/anir16293/Deep-Lesion/master/DL_info.csv"
+# Path to images and category labels in data dir
+_DATA_DIR = "Deeplesion-v0.1.0"
+_TRAIN_LABELS_FNAME = "train.csv"
+_VALIDATION_LABELS_FNAME = "valid.csv"
+
+_ANNOTATION_CATEGORY = collections.OrderedDict({
+    '1':'Bone', 
+    '2':'Abdomen', 
+    '3':'Mediastinum', 
+    '4':'Liver', 
+    '5':'Lung', 
+    '6':'Kidney', 
+    '7':'Soft tissue', 
+    '8':'Pelvis', 
+    '-1':'unknown',
+})
+
+
+class Deeplesion(tfds.core.GeneratorBasedBuilder):
+  """DeepLesion dataset."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "name":tfds.features.Text(),
+            "image":tfds.features.Image(encoding_format='png'),
+            "bbox":tfds.features.BBoxFeature(),
+            "label":tfds.features.ClassLabel(names=_ANNOTATION_CATEGORY.values()),
+        }),
+        # Homepage of the dataset for documentation
+        urls=[_URL],
+        citation=_CITATION,
+    )
+
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    path = dl_manager.manual_dir
+    imgs_path = os.path.join(path, _DATA_DIR)
+
+    if not tf.io.gfile.exists(imgs_path):
+      msg = ("You must download the dataset folder from {} \
+        manually and place it into {}.".format(_BOX_URL, image_path))
+      raise AssertionError(msg)
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=100,
+            gen_kwargs={
+                "imgs_path": imgs_path,
+                "csv_path": os.path.join(path, _TRAIN_LABELS_FNAME)
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=10,
+            gen_kwargs={
+                "imgs_path": imgs_path,
+                "csv_path": os.path.join(path, _VALIDATION_LABELS_FNAME)
+            },
+        ),
+    ]
+
+
+  def _generate_examples(self, imgs_path, csv_path):
+    """Yields examples."""
+    def _decode_name(slice_uid):
+      """decode name to get valid path
+
+        params:
+          slice_uid, str with format patient_study_series_slice.png
+        return:
+          valid path to find the slice
+      """
+      folder = '_'.join(slice_uid.split('_')[0:-1])
+      fname = slice_uid.split('_')[-1]
+      return os.path.join(folder, fname)
+    
+    with tf.io.gfile.GFile(csv_path) as csv_f:
+      pd = tfds.core.lazy_imports.pandas
+      data = pd.read_csv(csv_path).values
+
+    for idx, value in enumerate(data):
+      slice_uid, xmin, ymin, xmax, ymax, label = value
+      record = {
+          "name": slice_uid,
```

Comment:
Deep Lesion is a 3d dataset, add transforms in `_as_dataset` to help deal with this.

Reply:
> Deep Lesion is a 3d dataset, add transforms in `_as_dataset` to help deal with this.

working on it